cache/cache.go

```diff
@@ -20,7 +20,7 @@ type RemoteCacheClient interface {
 	// SetAsync enqueues an asynchronous operation to store a key into memcached.
 	// Returns an error in case it fails to enqueue the operation. In case the
 	// underlying async operation will fail, the error will be tracked/logged.
-	SetAsync(ctx context.Context, key string, value []byte, ttl time.Duration) error
+	SetAsync(key string, value []byte, ttl time.Duration) error
```

Comment:
Out of the scope of this PR, but something that might be good to look at while we're making breaking changes: is there a reason why the methods on `Cache` and `RemoteCacheClient` are named differently? eg. `Fetch` / `GetMulti` and `StoreAsync` / `SetAsync`

Reply:
Yes! Longer term I want to remove `RemoteCacheClient` entirely. It's an abstraction around Redis/Memcached which is then used by our even more general `cache.Cache` abstraction.